src/util/type.h

```diff
@@ -224,4 +224,8 @@ inline type_with_subtypest &to_type_with_subtypes(typet &type)
     for(type_with_subtypest::subtypest::iterator it=to_type_with_subtypes(type).subtypes().begin(); \
         it!=to_type_with_subtypes(type).subtypes().end(); ++it)
 
+/// Remove const qualifier from type (if any).
+/// Returns type as is if there is no const qualifier.
+typet remove_const(typet type);
```

Comment:
:bulb: I guess I'd probably expect this function to recurse... perhaps it should just be a static util function in this place where it is needed?

Reply:
I was modelling this after [`std::remove_const`](https://en.cppreference.com/w/cpp/types/remove_cv) which doesn't recurse

Reply:
Yeah, I'm not sure why this is a member function rather than a static?

Reply:
@chrisr-diffblue it is free standing, not a member. Thomas was suggesting moving it to goto harness where it was used at the time, but this is the sort of thing I could see being useful elsewhere as well and I figured we don't want to have too much proliferation of duplicated utility functions.

Reply:
Ah, sorry - I failed to read the diff context properly :-)